src/main/java/com/google/sps/data/DatastoreManagerInterface.java

```diff
@@ -0,0 +1,28 @@
+package com.google.sps.data;
+
+import java.util.Date;
+import com.google.appengine.api.datastore.Entity;
+import java.util.Optional;
+import java.util.List;
+
+/** Interface for the DatastoreManager class. */
+public interface DatastoreManagerInterface {
+    void addNewAttendee(String screenName, String sessionId);
+    void addNewInstance(String instanceName, Optional<String> sessionId);
+    void addNewSession(String sessionId,
+        Optional<String> screenNameOfController, Optional<String> ipOfVM);
+    void updateAttendee(String screenName, String sessionId);
+    void updateStateOfInstance(String instanceName, String stateOfInstance);
+    void updateSessionIdOfInstance(String instanceName, 
+        Optional<String> sessionId);
+    void updateControllerOfSession(String sessionId,
+        String screenNameOfController);
+    void updateIpOfVM(String sessionId, Optional<String> ipOfVM);
```

Comment:
Pro tip: You dont need to have a function for every type of thing that can be updated. Instead, you can do this:
void insertOrUpdateAttendee(AttendeeInterface attendee);
void insertOrUpdateInstance(InstanceInterface instance);
void insertOrUpdateSession(SessionInterface session);

If the thing being inserted already exists in the datastore, then the old one will be replaced with the new thing. if it doesnt, then a new entity will be added. 

Let the caller create the Session/Instance/Attendee objects. Your job is just to make sure it gets to the datastore. Your job isnt to support every permutation of update. So the caller will look like this for example:

void handleCreateSession() {
  SessionInterface session = new Session(....);
  datastoreManager.insertOrUpdateSession(session);
 ...
 ...
}

void handleChangeSessionVM() {
  SessionInterface session = datastoreManager.getSession(...);
  session.setIpOfVm(...);
  datastoreManager.insertOrUpdateSession(session);
}